### PR TITLE
Core/SAI: Add extra action param 7

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Add another parameter due to some npcs are using 7 random emotes, but atm you can only add 6

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -2553,6 +2553,7 @@ SmartScriptHolder SmartScript::CreateSmartEvent(SMART_EVENT e, uint32 event_flag
     script.action.raw.param4 = action_param4;
     script.action.raw.param5 = action_param5;
     script.action.raw.param6 = action_param6;
+    script.action.raw.param6 = action_param7;
```
Comment (jackpoz) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:2556]: you copied param6 instead of param7. are you sure you tested ingame this PR ?

**Comment by aquadeus** [comment]
oops i changed in localhost but i forget to upload the change

**Comment by mdX7** [comment]
you might also want to change `SMART_ACTION_PARAM_COUNT` then

**Comment by aquadeus** [comment]
> you might also want to change `SMART_ACTION_PARAM_COUNT` then

done

**Comment by Lordron** [comment]
bloating params is not a healthy way to do it. you can use timed actions for as many emotes as you want. What if we will discover an npc with 8 or more emotes?

**Comment by mdX7** [comment]
> bloating params is not a healthy way to do it. you can use timed actions for as many emotes as you want. What if we will discover an npc with 8 or more emotes?

in that case we could create a table where we reference all related values under a certain id and just reference the id here, that way we'd be future proof. CMaNGOS also does it in that way iirc

**Comment by Lordron** [comment]
> > bloating params is not a healthy way to do it. you can use timed actions for as many emotes as you want. What if we will discover an npc with 8 or more emotes?
> 
> in that case we could create a table where we reference all related values under a certain id and just reference the id here, that way we'd be future proof. CMaNGOS also does it in that way iirc

and dump action_param1-6 to it leaving with 1 single actionParamId.

**Comment by Shauren** [comment]
I've seen up to 8 params max for blizz AI_GROUP_ACTION

**Comment by Shauren** [comment]
This was not tested AT ALL, the most imporatant piece is missing - accessing db query result

**Pull request merged**